### PR TITLE
GET /v1/accounts/:accountId/sources/:sourceId endpoint

### PR DESCRIPTION
**Pull request by mlaflamm**

### CODE REVIEW

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-source-pr-1** [here](http://e2-source.jx-lvlstudio-e2-source-pr-1.e2-dev.com)

File: contract/openapi.yaml [hunk 1]
```diff
@@ -1,13 +1,39 @@
...
 servers:
   - url: http://localhost:3000
 paths:
+  /v1/accounts/{accountId}/sources/{sourceId}:
+    get:
+      security:
+        - OAuth2: []
+      tags:
+        - sources
+      summary: fetch full movie
```
Comment (davidmdm) [inline contract/openapi.yaml]: i like movies

File: src/env.ts [hunk 2]
```diff
@@ -65,12 +66,13 @@ export class Environment {
...
-      level: (getOsEnv('LOG_LEVEL') as LogLevel) || 'info',
-    });
-
+    const authBaseUrl = getOsEnv('AUTH_BASE_URL') || 'https://e2-auth.staging.e2-dev.com';
     this.auth = Object.freeze({
-      route: getOsEnv('AUTH_BASE_URL'),
+      issuer: authBaseUrl,
+      authorizationUrl: authBaseUrl + '/authorize',
+      tokenUrl: authBaseUrl + '/token',
+      clientId: getOsEnv('E2_CLIENT_ID') || 'e2-source-service',
```
Comment (davidmdm) [inline src/env.ts]: configmap? secrets.yaml?

File: src/libs/services/source.service.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import { ContentstoreConsumerFactory } from './contentstore.consumer';
+
+export type GetContentOptions = {
+  accountId: string;
+  sourceId: string;
+  params: Dictionary<string>;
+};
+
+const resolveQueryParams = (query: ItemQuery, params: Dictionary<string>): ItemQuery => {
+  const mapValue = (value: any): any => {
```
Comment (davidmdm) [inline src/libs/services/source.service.ts]: mapValue and queryReducer are too pure functions that don't need to be within closure. I would right them outside and maybe prefix them with underscores.

File: src/libs/services/source.service.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        const [item] = await indexConsumer.queryItems({ ...query, limit: 1 });
+        return item;
+      }
+    })();
+
+    // TODO: optimize to run in parallel of root if children doesn't depend on root!
+    const childrenIds = await (async () => {
+      if (definition.items) {
+        const childrenParams = { ...params, ROOT_ID: rootId };
+        debug('Resolving %j children %j', { ...opts, params: childrenParams }, definition.items.query);
```
Comment (davidmdm) [inline src/libs/services/source.service.ts]: what happens if root_id is undefined??? patate

File: src/loaders/source.loader.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { MicroframeworkLoader, MicroframeworkSettings } from 'microframework-w3tec';
+import { ContainerInstance } from 'typedi';
+
+import { Environment } from '../env';
+import { SourceRepository } from '../libs/repositories/source.repository';
+
+export const sourceLoader = (env: Environment, container: ContainerInstance): MicroframeworkLoader => {
+  return (settings: MicroframeworkSettings | undefined) => {
+    const definitionsPath = 'src/libs/repositories/definitions';
```
Comment (davidmdm) [inline src/loaders/source.loader.ts]: Better be temporary. Especially since you are reading from "src".

**Comment by e2-cloud** [comment]
:star: PR built and available in a preview environment **lvlstudio-e2-source-pr-1** [here](http://e2-source.jx-lvlstudio-e2-source-pr-1.e2-dev.com)

**Pull request merged**